util/artifacts_tar.go

```diff
@@ -26,6 +28,10 @@ func BuildArchive(ctx context.Context, tarWriter *tar.Writer, rootPath string, i
 	done := make(chan bool)
 	errChan := make(chan error)
 	go func(inputChan <-chan ArchiveContentFile) {
+		defer func() {
```

Comment:
should we also recover [here](https://github.com/evergreen-ci/evergreen/blob/de8b05081a12a47b76aa66dfd69b9533a9b2d496/util/artifacts.go#L43)?

Reply:
good catch! This won't catch our error because we know it happens within the loop but we may as well just fix it too